data/state/account_test.go

```diff
@@ -242,3 +243,104 @@ func TestAccount_SetCodeHashWithJournal(t *testing.T) {
 	assert.Equal(t, 1, journalizeCalled)
 	assert.Equal(t, 1, saveAccountCalled)
 }
+
+func TestAccount_SetOwnerAddressWithJournal(t *testing.T) {
+	t.Parallel()
+
+	count := 0
+	tracker := &mock.AccountTrackerStub{
+		JournalizeCalled: func(entry state.JournalEntry) {
+			count++
+		},
+		SaveAccountCalled: func(accountHandler state.AccountHandler) error {
+			count++
+			return nil
+		},
+	}
+
+	acc, _ := state.NewAccount(&mock.AddressMock{}, tracker)
+
+	owner := []byte("owner")
+	err := acc.SetOwnerAddressWithJournal(owner)
+	require.Nil(t, err)
+	require.Equal(t, 2, count)
+	require.Equal(t, acc.OwnerAddress, owner)
+}
+
+func TestAccount_AddAndClaimDeveloperRewards(t *testing.T) {
+	t.Parallel()
+
+	tracker := &mock.AccountTrackerStub{
+		JournalizeCalled: func(entry state.JournalEntry) {
+		},
+		SaveAccountCalled: func(accountHandler state.AccountHandler) error {
+			return nil
+		},
+	}
+
+	acc, _ := state.NewAccount(&mock.AddressMock{}, tracker)
+	owner := []byte("owner")
+	_ = acc.SetOwnerAddressWithJournal(owner)
+
+	devRewards := big.NewInt(100)
+	err := acc.AddToDeveloperReward(devRewards)
+	require.Nil(t, err)
+
+	rewards, err := acc.ClaimDeveloperRewards([]byte("wrongAddr"))
+	require.Nil(t, rewards)
+	require.Equal(t, state.ErrOperationNotPermitted, err)
+
+	rewards, err = acc.ClaimDeveloperRewards(owner)
+	require.Nil(t, err)
+	require.Equal(t, devRewards, rewards)
+	require.Equal(t, big.NewInt(0), acc.DeveloperReward)
+}
+
+func TestAccount_ChangeOwnerAddress(t *testing.T) {
+	t.Parallel()
+
+	tracker := &mock.AccountTrackerStub{
+		JournalizeCalled: func(entry state.JournalEntry) {
+		},
+		SaveAccountCalled: func(accountHandler state.AccountHandler) error {
+			return nil
+		},
+	}
+
+	acc, _ := state.NewAccount(mock.NewAddressMock(), tracker)
+	owner := []byte("owner")
+	_ = acc.SetOwnerAddressWithJournal(owner)
+
+	newOwnerAddr := []byte("newOwner")
```

Comment:
you can make an args struct which will hold different input data and the expected error (or nil). follow `TestSecondsToHourMinSec_ShouldWork` that you wrote for example

Reply:
i know what you mean but i will let it as it is because there just 3 scenarios here